LFS-473 / LFS-476: As an administrator I can specify a limit for the number of forms per subject in the Questionnaire designer

nela-jankechova opened the pull request:
This PR allows the administrator to add or change the `maxPerSubject` field of the questionnaire in the UI. However, the functionality for actually limiting the forms based on this field (LFS-473) is not complete. Therefore, changing this field will currently not affect the forms. This PR is based on LFS-172.
[Jira Link](https://phenotips.atlassian.net/browse/LFS-476)

nela-jankechova committed:
LFS-473_LFS-476: Create JSON file for properties

nela-jankechova committed:
LFS-473_LFS-476: Update Questionnaire, Edit Dialog

sdumitriu commented on modules/data-entry/src/main/frontend/src/questionnaire/EditDialog.jsx:
```diff
@@ -41,10 +41,11 @@ import Fields from './Fields'
 let EditDialog = (props) => {
-  let [ questionJSON, setQuestionJSON ] = useState(require('./Question.json'));
-  let [ sectionJSON, setSectionJSON ] = useState(require('./Section.json'));
+  let [ questionJSON ] = useState(require('./Question.json'));
```
This looks strange to me. Why is a dependency used as state?

sdumitriu commented on modules/data-entry/src/main/frontend/src/questionnaire/Questionnaire.jsx:
```diff
@@ -264,4 +281,4 @@ let AnswerOption = (props) => {
 AnswerOption.propTypes = {
   data: PropTypes.object.isRequired
-};
+ };
```
Revert this.
```suggestion
};
```

nela-jankechova committed:
Update modules/data-entry/src/main/frontend/src/questionnaire/Questionnaire.jsx

Co-authored-by: Sergiu Dumitriu <sdumitriu@users.noreply.github.com>

sashaandjic commented:
I can define the number of forms per subject, however whenever I create a new for I do not get prompted to select/create subject.

OrdiNeu committed:
LFS-473 / LFS-476: As an administrator I can specify a limit for the
number of forms per subject in the Questionnaire designer

Remove unused json

OrdiNeu committed:
LFS-473 / LFS-476: As an administrator I can specify a limit for the
number of forms per subject in the Questionnaire designer
Fix bugs that appeared during the refactor

sashaandjic commented:
tested;  I was able to define the maximum number of forms per subject
The maximum number of forms will get enforced by LFS-473 / LFS-474, LFS-475, LFS-477, LFS-478 PR.

sashaandjic reviewed (approved):
tested
 I was able to define the maximum number of forms per subject
The maximum number of forms will get enforced by https://github.com/ccmbioinfo/lfs/pull/239 PR.

sdumitriu reviewed (approved):
Possible improvement: limit the range of the setting to be >= 0.
